source/hp/dof_handler.cc

```diff
@@ -609,7 +617,7 @@ namespace internal
               types::global_dof_index cache_size = 0;
               for (typename HpDoFHandler<dim,spacedim>::active_cell_iterator
                    cell=dof_handler.begin_active(level);
-                   cell!=dof_handler.end_active(level); ++cell)
+                   cell!=end_active_cache[level]; ++cell)
```

Comment:
It's not wrong to do it this way, but couldn't you have just defined an iterator `endc` before the loop and compared with this? I suppose the run-time is this long because we evaluate `end_active(level)` for every iteration of the loop, right?